app/helpers/metadata_helper.rb

```diff
@@ -115,7 +122,19 @@ def metadata_template_csv_helper(project_id, new_sample_names)
             elsif project.nil?
               generate_metadata_default_value(field, sample[:host_genome_name])
             else
-              sample[:metadata][field.name] ? sample[:metadata][field.name].raw_value : nil
+              datum = sample[:metadata][field.name]
+              if datum
+                base_type = datum.metadata_field.base_type
+
+                # rubocop:disable Metrics/BlockNesting
+                if base_type == Metadatum::LOCATION_TYPE
+                  # Special-case for Location types
+                  val = datum.validated_value
+                  val.is_a?(Hash) ? val[:name] : val
+                else
+                  datum.raw_value
```

Comment:
I feel like this code block above would better as a method of `Metadatum`, possibly overriding `raw_value`. 

Is the special case relevant to CSV only or other potential call sites?

Reply:
That's a good suggestion 👍moved it to `Metadatum#csv_template_value`. It is mostly custom for CSV to have a CSV-friendly string value instead of formatting a date or a hash of location info